Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -779,6 +779,70 @@ public struct Timestamp: BSONValue, Equatable, Codable {
     }
 }
 
+enum BSONEqualsError: Error { case InvalidArrayArgument(String) }
+
+
+// See https://github.com/realm/SwiftLint/issues/461
+// swiftlint:disable cyclomatic_complexity
+/**
+ *  A helper function to test equality between two BSONValues. This function tests for exact BSON equality.
+ *  This means that differing types with equivalent value are not equivalent.
+ *
+ *  e.g.
+ *      4.0 (Double) != 4 (Int)
+ */
+func bsonEquals(lhs: BSONValue, rhs: BSONValue) throws -> Bool {
```

Comment:
TIL; done in f6f12c3.

This is a little weird because it was still accessible in the tests, which made me think that this was `public` by default. However, I see now that this is due to the `@testable import`!

Awesome new tidbit I learned about Swift.